packages/core/http/core-http-router-server-internal/src/router.ts

```diff
@@ -305,6 +337,8 @@ export class Router<Context extends RequestHandlerContextBase = RequestHandlerCo
       return response;
     }
 
+    emit?.onPostValidation(kibanaRequest, kibanaRequest.route.options);
```

Comment:
I was thinking about this... Maybe we should also emit in this in cases where validation fails... WDYT?

Same "fix" should be applied to versioned router.